src/test/java/ru/ayaz/UserMessageHandlerTest.java

```diff
@@ -0,0 +1,123 @@
+package ru.ayaz;
+
+import org.junit.Before;
+import org.junit.Test;
+
+import java.io.BufferedReader;
+import java.io.IOException;
+import java.io.PrintWriter;
+
+import static junit.framework.TestCase.assertEquals;
+import static junit.framework.TestCase.assertTrue;
+import static org.mockito.Mockito.*;
+
+public class UserMessageHandlerTest {
+
+    private UserMessageHandler messageHandler;
+
+    @Before
+    public final void before() {
+        messageHandler = new UserMessageHandler(500);
+    }
+
+    @Test
+    public void shouldBeAbleToEnqueueMessage() throws InterruptedException {
+        UserMessage someMessage = new UserMessage("Ayaz", "Hello");
+
+        messageHandler.enqueueMessage(someMessage);
+
+        assertTrue(messageHandler.queueContainsMessage(someMessage));
+    }
+
+    @Test
+    public void shouldBeAbleToGetEnqueuedMessage() throws InterruptedException {
+        UserMessage someMessage = new UserMessage("Ayaz", "Hello");
+        messageHandler.enqueueMessage(someMessage);
+
+        UserMessage gottenMessage = messageHandler.takeMessage();
+
+        assertEquals(someMessage, gottenMessage);
+    }
+
+
+    @Test
+    public void shouldInvokePrintUnknownCommandWhenMalformedCommandTaken() throws InvalidNicknameException, InterruptedException {
+        User user = new User();
+        user.setNickname("Ayaz");
+        UserMessage commandMessage = new UserMessage("Ayaz", "#quiit");
+        PrintWriter writer = mock(PrintWriter.class);
+        UserMessageHandler spyMessageHandler = spy(messageHandler);
+        spyMessageHandler.addToUsersMap(user);
+        spyMessageHandler.addToWritersMap(user.getNickname(), writer);
```

Comment:
After i summed up all architecture issues some troubles appeared: If i will encapsulate writer in UserHandler, i will not be able to write anything, because UserHandler supposed to read only.I think i should have only UserHandler and a simple one or two row flushing command in UserHandler. It means that UserHandler only reads and write but doesnt process and execute anything.